### PR TITLE
feat: test tracer

### PR DESCRIPTION
**Pull request by loocapro**
Implementing tracer for test tracer in order to remove init_test_tracer function
Updating book with new cli args

### CODE REVIEW

File: crates/rpc/rpc-builder/tests/it/http.rs [hunk 1]
```diff
@@ -514,7 +515,7 @@ async fn test_call_otterscan_functions_http_and_ws() {
 // <https://github.com/paradigmxyz/reth/issues/5830>
 #[tokio::test(flavor = "multi_thread")]
 async fn test_eth_logs_args() {
-    reth_tracing::init_test_tracing();
+    TestTracer.init().unwrap();
```
Comment (mattsse) [inline crates/rpc/rpc-builder/tests/it/http.rs:518]: I'd like to keep the function but we can make this the body of the init_test_tracing function

File: crates/tracing/src/test_tracer.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+use crate::Tracer;
+
+///  Initializes a tracing subscriber for tests.
+///
+///  The filter is configurable via `RUST_LOG`.
+///
+///  # Note
+///
+///  The subscriber will silently fail if it could not be installed.
+#[derive(Debug)]
```
Comment (mattsse) [inline crates/tracing/src/test_tracer.rs:13]:
```suggestion
#[derive(Debug, Clone, Default)]
#[non_exhaustive]
```

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/tracing/src/lib.rs [hunk 3]
```diff
@@ -223,8 +226,5 @@ impl Tracer for RethTracer {
 ///
 ///  The subscriber will silently fail if it could not be installed.
 pub fn init_test_tracing() {
-    let _ = tracing_subscriber::fmt()
-        .with_env_filter(EnvFilter::from_default_env())
-        .with_writer(std::io::stderr)
-        .try_init();
+    TestTracer.init().unwrap();
```
Comment (mattsse) [inline crates/tracing/src/lib.rs:229]: this should not panic (equivalent to previous behaviour)
```suggestion
   let _ = TestTracer::default().init();
```

**Review by mattsse — CHANGES_REQUESTED** [review]
we can discard all the new imports because if we keep the fn these don't have any effect.

**Review by mattsse — APPROVED** [review]
lgtm, pending @shekhirin

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**